### DISCUSSION TITLE
how make all single color pixels with fuzz which partial transparent to transparent or not transparent?

### DISCUSSION THREAD

**Discussion by markonop93** [ImageMagick/ImageMagick#6908] [Help] [answered]
### ImageMagick version

7.1.1-21

### Operating system, version and so on

Windows 11

### Description

i need to make all partial transparent orange pixel , with fuzz 20% , transparent. or not transparent if easier

so change opacity to 0 or 100 for color hex #FF6D2D only and all other fuzzy color wich match -fuzz 20%

how to do ??

**Comment by fmw42** [comment]
Please post an example image. Sorry, I do not fully understand your question.  You have an image with some transparency. You want to change one color with some tolerance (fuzz) that is opaque to transparence. Is this correct?  If not, please explain further.

**Reply by markonop93** [reply]
sorry for bad language. i need change only pixel of one color (example orange hex #FF6D2D) with fuzz 20% to color and only if the pixels are more or less transparent.

so
pixel orange and has transparent - make full transparent
pixel orange and dont have any transparent - dont change
pixel similar to orange with fuzz and has transparent - make full transparent

image is private but i try to find or make other example later

**Reply by fmw42** [reply]
Please post an example image.

**Reply by fmw42** [reply]
try this

Your hex color is:

FF6D2D => 255,109,45. (hex to rgb)

then normalize to range 0 to 1

255/255=1
109/255=0.427
45/255=0.176

So

`magick image.png -fuzz 20% -fx "(u.a>0 && u.r==1 && u.g==0.427 && u.b==0.176)?u.a==0:u" result.png`

If all your conditions are met, then set alpha to 0 (u.a==0), otherwise keep image (u) unchanged

**Comment by snibgo** [comment, marked as answer]
fmw42's solution won't work because `-fx` is not sensitive to `-fuzz`, and because `u.a==0` doesn't set u.a to zero, it merely *tests* whether u.a is zero. Instead, I suggest:

```
magick image.png -channel A -fx "(u.a<1 && abs(u.r-1)<0.2 && abs(u.g-0.427)<0.2 && abs(u.b-0.176)<0.2) ? 0 : u.a" +channel result.png
```
Note that this processes only the alpha channel.

This is difficult to test without an example image. @markonop93, please supply an example input.

**Reply by markonop93** [reply]
thanks this work

question appears - is easy way to use HEX value rather ?? for example i change color a bit here
![test](https://github.com/ImageMagick/ImageMagick/assets/122524476/518a4265-c9bd-4e88-8704-e2e3014ce457)

is any fx func that help with hex instead manual math ?? eg 0xFF and not 1

**Comment by snibgo** [comment]
`#` colours can be used in `-fx`. For example:
```
D:\temp>%IMG7%magick xc: -fx "#123 *2" txt:
# ImageMagick pixel enumeration: 1,1,0,65535,srgb
0,0: (8738,17476,26214)  #222244446666  srgb(34,68,102)
```

**Reply by markonop93** [reply]
yes i mean in your example replace 1, 0.427 and other

is possible ??

**Comment by snibgo** [comment]
`#123` is a colour.

`1` in my example above is a (decimal) number.

Numbers are not the same as colours.

`-fx` does not accept hex numbers. Even if it did, decimal `1` in hex would be `1`.

**Reply by markonop93** [reply]
i know

i ask to replace for easy use in script so i separate for example ff7709 to ff 77 09

but if no possible then i do convert
